instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/TracingClientInterceptor.java

```diff
@@ -109,16 +109,17 @@ public void start(Listener<RESPONSE> responseListener, Metadata headers) {
 
     @Override
     public void sendMessage(REQUEST message) {
+      Span span = Span.fromContext(context);
+      Attributes attributes =
+          Attributes.of(MESSAGE_TYPE, SENT, MESSAGE_ID, MESSAGE_ID_UPDATER.incrementAndGet(this));
+      span.addEvent("message", attributes);
+
       try (Scope ignored = context.makeCurrent()) {
         super.sendMessage(message);
       } catch (Throwable e) {
         instrumenter.end(context, request, Status.UNKNOWN, e);
         throw e;
       }
-      Span span = Span.fromContext(context);
-      Attributes attributes =
-          Attributes.of(MESSAGE_TYPE, SENT, MESSAGE_ID, MESSAGE_ID_UPDATER.incrementAndGet(this));
-      span.addEvent("message", attributes);
```

Comment:
I'm thinking (guessing) that it's better to only emit the event when the message is successfully sent (or if we're going to emit it when it fails to note the failure somehow in the event that is emitted)

I think it's fine to loosen the assertion ordering constraint